src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int INDEX_HOURS = 2;
+    private static final int SALARY_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder("Report for period ");
+        result.append(dateFrom).append(" - ").append(dateTo);
+        if (dateFrom == null || dateFrom.isEmpty() || dateTo == null || dateTo.isEmpty()
+                || names.length == 0 || data.length == 0) {
+            return "0";
```

Comment:
rather than return null or any other dummy result it's better to use proper exception with adequate message
```suggestion
            throw new IllegalArgumentException("input is not valid");
```